### PR TITLE
Updating README to include flow description and deploy using OLM

### PR DESCRIPTION
**Pull request by galletti94**
Thought we could use a section to explain the flow of the operator (even if subject to change in the future). Also including a description of how to deploy marketplace using OLM (for when @awgreene merges the CSV).

Let me know if you think this is needed - happy to change the wording, style etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,14 +22,26 @@ $ export REGISTRY=<SOME_REGISTRY> \
 ```
 2. Update the `deploy/operator.yaml` to pull the Marketplace Operator image you just pushed. You should update the `spec.template.spec.containers[0].image` field with the `$REGISTRY/$NAMESPACE/$REPOSITORY:$TAG` value.
 
-## Deploying the Marketplace Operator
+## Using the Marketplace Operator
 
-### In an OKD Cluster
+### Description
+
+The marketplace operator manages two CRDs: [operatorsource](./deploy/operatorsource.crd.yaml) and [catalogsourceconfig](./deploy/catalogsourceconfig.crd.yaml). When an operatorsource CR is created in the same namespace as where the marketplace operator is running, the marketplace operator will download manifests stored in the external registry specified in this operatorsource CR (for now, please see documentation about using quay's appregistry API). For an example of this operatorsource CR please see the [examples](./deploy/examples/) folder.
```
Comment (aravindhp) [inline README.md]: Please use 'OperatorSource` and `CatalogSource` etc when mentioning CRs.

File: README.md [hunk 2]
```diff
@@ -22,14 +22,26 @@ $ export REGISTRY=<SOME_REGISTRY> \
...
+#### In an OKD Cluster
 ```bash
-$ oc apply -f deploy
+$ oc apply -f deploy/
 ```
 
-### In a Kubernetes Cluster
+#### In the UI
+
+When deploying OLM, include the contents of the CSV file located in the [deploy](./deploy/) folder in the rh-operators configmap shipped with the latest version of OLM. Follow the remaining instructions for deploying OLM. Finally, navigate to the `package manifest` navigation tab in the `Operators` section, find the marketplace operator, and follow the steps after clicking `Create Subscription` in the OLM UI to deploy the operator.
```
Comment (aravindhp) [inline README.md]: This should go in only after we have added the CSV to repo. Moreover once that is in and is part of rh-operators, you only need to mention how to subscribe to it. It should also be called out that this is a OpenShift UI.

File: README.md [hunk 3]
```diff
@@ -22,14 +22,26 @@ $ export REGISTRY=<SOME_REGISTRY> \
...
+
+#### In a Kubernetes Cluster
 ```bash
-$ kubectl apply -f deploy
+$ kubectl apply -f deploy/
```
Comment (aravindhp) [inline README.md]: This would need to be updated if we want users to go the CSV route.

**Comment by galletti94** [comment]
@aravindhp I think I addressed all the issues you raised - let me know if there is anything else you'd like to tweak

**Comment by galletti94** [comment]
@aravindhp is this good to go?

**Pull request merged**